platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongList.java

```diff
@@ -297,12 +348,23 @@ protected final void writeHeader(final FileChannel fc) throws IOException {
     protected abstract long lookupInChunk(final long chunkIndex, final long subIndex);
 
     /**
-     * Current implementation is no-op.
+     * Calls implementation specific {@link LongList#onUpdateMinValidIndex(long)}
+     * and then updates {@code minValidIndex} parameter
      *
      * @param newMinValidIndex minimal valid index of the list
      */
-    public void updateMinValidIndex(final long newMinValidIndex) {
-        // no-op
+    public final void updateMinValidIndex(final long newMinValidIndex) {
+        onUpdateMinValidIndex(newMinValidIndex);
+        minValidIndex.set(newMinValidIndex);
+    }
+
+    /**
+     * An action that has to be taken before update of the min valid index
+     *
+     * @param newMinValidIndex min valid index
+     */
+    protected void onUpdateMinValidIndex(final long newMinValidIndex) {
```

Comment:
It might be more flexible to have this return a long, so the subclass has the opportunity to change the value before it's set, rather than only taking some action with no possibility of affecting the value.

Reply:
Not sure if I understand what you mean. The goal of this method is to decouple actual update of `minValidIndex` and the rest of the logic that has to happen along with the update . By design, the update of the index is the last thing that has to happen. New `minValidIndex` determined by the client code and is not supposed to be calculated by the subclasses.

Reply:
I was just thinking that if you allow the subclass to take action when the number changes, then a subclass might also want to update that min index (for instance, perhaps the subclass defers the change and updates the value again after some condition is met).
I don't have a clear use-case up front, just thinking it might be useful.